app/Http/Controllers/Api/ReportbackController.php

```diff
@@ -23,15 +24,28 @@ public function __construct(ReportbackService $reportbackService)
     }
 
     /**
-     * Store a newly created resource in storage.
-     * @todo upsert reportbacks by default.
+     * Store a newly created resource in storage or
+     * update a reportback if it already exists.
+     *
      * @param  \Illuminate\Http\Request  $request
      * @return \Illuminate\Http\Response
      */
-    public function create(Request $request)
+    public function store(ReportbackRequest $request)
     {
-        $reportback = $this->reportbackService->create($request->all());
+        // @TODO - instead should probably just have a method that gets northstar_id by default from a drupal_id if that is the only thing provided and then use that to find the reportback.
+        $userId = $request['northstar_id'] ? $request['northstar_id'] : $request['drupal_id'];
+        $type = $request['northstar_id'] ? 'northstar_id' : 'drupal_id';
 
-        return $this->item($reportback);
+        $reportback = $this->reportbackService->exists($request['campaign_id'], $request['campaign_run_id'], $userId, $type);
+
+        if (! $reportback) {
+            $reportback = $this->reportbackService->create($request->all());
```

Comment:
The idea is that the service would handle all business logic around creating a reportback and the repository handles all database level transactions. 

So for example in this iteration of the project, when a reportback gets created we send it back to phoenix. The service would handle connecting to the phoenix service and doing that request but won't have to worry about all the logic around adding a reportback to the db (which involves updating a few tables). 

Or if you wanted to like, send an event out that a reportback was created, you would do it in the service class.